src/condition.cpp

```diff
@@ -1741,10 +1741,18 @@ conditional_t::func f_map_in_city( const JsonObject &jo, std::string_view member
 {
     str_or_var target = get_str_or_var( jo.get_member( member ), member, true );
     return [target]( dialogue const & d ) {
-        tripoint_abs_ms target_pos = tripoint_abs_ms( tripoint::from_string( target.evaluate( d ) ) );
-        city_reference c = overmap_buffer.closest_city( project_to<coords::sm>( target_pos ) );
-        c.distance = rl_dist( c.abs_sm_pos, project_to<coords::sm>( target_pos ) );
-        return c && c.get_distance_from_bounds() <= 0;
+        tripoint_abs_omt target_pos = project_to<coords::omt>( tripoint_abs_ms( tripoint::from_string(
+                                          target.evaluate( d ) ) ) );
+
+        // TODO: Remove this in favour of a seperate condition for location z-level that can be used in conjunction with this map_in_city as needed
+        point_abs_om overmap_pos;
+        tripoint_om_omt potential_city_tile;
+        std::tie( overmap_pos, potential_city_tile ) = project_remain<coords::om>( target_pos );
+        if( potential_city_tile.z() < -1 ) {
```

Comment:
I don't think any projection will change the z position
```suggestion
        if( target_pos.z() < -1 ) {
```

Reply:
Oops yes that's much less ugly ;D